ansible/terraform/aws/setup.tf

```diff
@@ -25,46 +36,124 @@ data "aws_ami" "ubuntu" {
   owners = ["099720109477"] # Canonical
 }
 
+data "aws_vpc" "sf_vpc" {
+  id = var.vpc_id
+}
+
+resource "aws_security_group" "sf_allow_ssh" {
+  name        = "sf_allow_ssh"
+  description = "Allow SSH inbound traffic"
+  vpc_id      = data.aws_vpc.sf_vpc.id
+
+  ingress {
+    description = "SSH"
+    from_port   = 22
+    to_port     = 22
+    protocol    = "tcp"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  tags = {
+    Name = "sf_allow_ssh"
+  }
+}
+
+resource "aws_security_group" "sf_allow_api" {
+  name        = "sf_allow_api"
+  description = "Allow SF API inbound traffic"
+  vpc_id      = data.aws_vpc.sf_vpc.id
+
+  ingress {
+    description = "SF API"
+    from_port   = 13000
+    to_port     = 13000
+    protocol    = "tcp"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  tags = {
+    Name = "sf_allow_api"
+  }
+}
+
+resource "aws_security_group" "sf_allow_vxlan" {
+  name        = "sf_allow_vxlan"
+  description = "Allow VXLAN traffic"
+  vpc_id      = data.aws_vpc.sf_vpc.id
+
+  ingress {
+    description = "VXLAN"
+    from_port   = 8472
+    to_port     = 8472
+    protocol    = "udp"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  ingress {
+    description = "VXLAN"
+    from_port   = 4789
+    to_port     = 4789
+    protocol    = "udp"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  tags = {
+    Name = "sf_allow_vxlan"
+  }
+}
+
 resource "aws_instance" "sf_1" {
   ami           = data.aws_ami.ubuntu.id
   instance_type = "c5d.metal"
-  key_name      = "mikal-aws"
+  key_name      = var.ssh_key_name
   root_block_device {
     delete_on_termination = true
     volume_size           = 20
   }
   tags = {
     Name = "sf-1"
   }
-  vpc_security_group_ids = ["sg-0ff088c6b3e980ffd", "sg-0008a21805a524651"]
+  vpc_security_group_ids = [
+    aws_security_group.sf_allow_ssh.id,
+    aws_security_group.sf_allow_ssh.id,
```

Comment:
Trying to comment about line 118 aws/setup.tf - is there no reference included?

This line I think should by allow_api. Same applies for sf_2 and 3.